docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,579 @@
+
+In the following diagram, you can also see how the aggregator services work with their related API Gateways.
+
+![](./media/image37.png)
+
+**Figure 8-36.** eShopOnContainers architecture with aggregator services
+
+The following image is zooming in further, so you can notice how for the “Shopping” business area, the client apps could be improved by reducing chattiness with microservices when using those aggregator services under the realm of the API Gateways. 
+
+ ![](./media/image38.png)
+
+**Figure 8-37.** Zoom in vision of the Aggregator services
+
+You can notice how when the diagram shows the possible requests coming from the API Gateways it can get pretty complex. Although you can see how the arrows in blue would be simplified, from a client apps perspective, when using the aggregator pattern by reducing chattiness and latency in the communication, ultimately significantly improving the user experience for the remote apps (mobile and SPA apps), especially. 
+
+In the case of the “Marketing” business area and microservices, it is a very simple use case so there was no need to use aggregators, but it could also be possible, if needed.
+
+### Authentication and authorization in Ocelot API Gateways
+
+In an Ocelot API Gateway you can sit the authentication service, such as an ASP.NET Core Web API service using [IdentityServer](http://identityserver.io/) providing the auth token, either out or inside the API Gateway.
+
+Since in eShopOnContainers is using multiple API Gateways with boundaries based on BFF and business areas, the Identity/Auth service is left out of the API Gateways, as highlighted in yellow in the following diagram.
+
+ ![](./media/image39.png)
+
+**Figure 8-38.** Position of the Identity service in eShopOnContainers
+
+However, Ocelot also supports to sit the Identity/Auth microservice within the API Gateway boundary, as in this other diagram.
+
+ ![](./media/image40.png)
+
+**Figure 8-39.** Authentication in Ocelot API Gateway
+
+Since in eShopOnContainers we have split the API Gateway into multiple BFF (Backend for Frontend) and business areas API Gateways, another option would had been to create an additional API Gateway for cross-cutting concerns. That choice would be fair in a more complex microservice based architecture with multiple cross-cutting concerns microservices. But sin we have just one cross-cutting concerns in eShopOnContainers it was decided to just handle the security service out of the API Gateway realm, for simplicity’s sake.
+
+In any case, the authentication module of Ocelot API Gateway will be visited at first when trying to use any secured microservice (if secured at the API Gateway level). That will redirect to visit the Identity or auth microservice to get the access token so you can access the protected services with the access_token.
```

Comment:
of Ocelot -> of the Ocelot

should the sentence in parentheses be outside and introducing the sentence perhaps?

can you remove future tense?

Reply:
👍